### DISCUSSION TITLE
How to check if input has focus?

### DISCUSSION THREAD

**Discussion by jochemgruter** [alpinejs/alpine#1951] [1. Help]
I want to show a div underneath an input text field, only when it has focus. How can I do this?

I tried something like this, but its not working:
```
<div x-data>
    <input type="text" name="query" x-ref="query" /> 
    <div x-show="$refs.query === document.activeElement">Test</div>
</div>
```

**Comment by SimoTod** [comment]
In v3, only properties in x-data are reactive by default. You can do something like this instead 
```html
<div x-data="{show: false}">
    <input type="text" name="query" @focus="show = true" @blur="show = false"/>
    <div x-show="show">Test</div>
</div>
```

**Comment by quasipickle** [comment]
You can do this with just CSS with the :focus state and sibling selectors: https://codepen.io/quasipickle/pen/LYyvrzv

**Comment by erikscheepers** [comment]
A related issue: what if I want to see 'Test' when either of the two inputs has focus? Currently, the code below doesn't work. When the second input gets focus also the "Don't show" text is visible.

```html
<div x-data="{show: false}">
    <input type="text" name="thing_1" @focus="show = true" @blur="show = false"/>
    <input type="text" name="thing_2" @focus="show = true" @blur="show = false"/>
    <div x-show="show">Test</div>
    <div x-show="! show">Don't show</div>
</div>
```

**Reply by Tim-Wils** [reply]
I think the problem is the order when focus and blur are fired after tabbing through the inputs. This is solved by setting the show-state on the $nextTick instead of immediately.
Working $nextTick example:
https://jsfiddle.net/8jv0wotq/
